app/jobs/irs_attempts_events_batch_job.rb

```diff
@@ -0,0 +1,34 @@
+class IrsAttemptsEventsBatchJob < ApplicationJob
+  queue_as :default
+
+  def perform(subject_timestamp = Time.zone.now - 1.hour, dir_path = "./attempts_api_output")
+    return nil unless IdentityConfig.store.irs_attempt_api_enabled
+    
+    events = IrsAttemptsApi::RedisClient.new.read_events(timestamp: subject_timestamp)
+    event_values = events.values.join("\r\n")
+   
+    decoded_key_der = Base64.strict_decode64(IdentityConfig.store.irs_attempt_api_public_key)
+    pub_key = OpenSSL::PKey::RSA.new(decoded_key_der)
+
+    result = IrsAttemptsApi::EnvelopeEncryptor.encrypt(
+      data: event_values, timestamp: subject_timestamp, public_key: pub_key,
+    )
+    
+    # write to a file and store on the disk until S3 is setup
+    begin
+      Dir.mkdir(dir_path) unless File.exists?(dir_path)
+
+      file = File.open("#{dir_path}/#{result.filename}", 'wb')
+      file.write(result.encrypted_data) 
+    rescue IOError => e
+      puts e
+    ensure
+      file.close unless file.nil?
+    end
+    #puts "Wrote to file: #{file.path}"
+    return file.path
+
+    # Write the file to S3 instead of whatever dir_path winds up being
+
+  end
+end
```

Comment:
![bad](https://thumbs.gfycat.com/FrighteningDifferentAmethystgemclam-size_restricted.gif)

This is missing a newline.

Reply:
So, I tried to add a space here, but it seems like github truncated it?